tornado/ioloop.py

```diff
@@ -173,15 +173,12 @@ async def main():
     def configure(
         cls, impl: "Union[None, str, Type[Configurable]]", **kwargs: Any
     ) -> None:
-        if asyncio is not None:
-            from tornado.platform.asyncio import BaseAsyncIOLoop
-
-            if isinstance(impl, str):
-                impl = import_object(impl)
-            if isinstance(impl, type) and not issubclass(impl, BaseAsyncIOLoop):
-                raise RuntimeError(
-                    "only AsyncIOLoop is allowed when asyncio is available"
-                )
+        from tornado.platform.asyncio import BaseAsyncIOLoop
+
+        if isinstance(impl, str):
+            impl = import_object(impl)
+        if isinstance(impl, type) and not issubclass(impl, BaseAsyncIOLoop):
+            raise RuntimeError("only AsyncIOLoop is allowed when asyncio is available")
```

Comment:
this message is somewhat misleading now - maybe it would be better to merge BaseAsyncIOLoop into IOLoop?

Reply:
We could merge BaseAsyncIOLoop into IOLoop, but it wouldn't change the need for this message. The `configure` method now exists only for backwards compatibility. Old code may still be calling `IOLoop.configure(AsyncIOLoop)` and we want that to continue to work, but it should still be an error to call `configure(TwistedIOLoop)` (for example).